debezium-core/src/main/java/io/debezium/transforms/HeaderToValue.java

```diff
@@ -0,0 +1,289 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.transforms;
+
+import static io.debezium.transforms.HeaderToValue.Operation.MOVE;
+import static java.lang.String.format;
+import static org.apache.kafka.connect.transforms.util.Requirements.requireStruct;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+import org.apache.kafka.common.config.ConfigDef;
+import org.apache.kafka.common.config.ConfigException;
+import org.apache.kafka.connect.connector.ConnectRecord;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.header.Header;
+import org.apache.kafka.connect.header.Headers;
+import org.apache.kafka.connect.transforms.Transformation;
+import org.apache.kafka.connect.transforms.util.SchemaUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.debezium.config.Configuration;
+import io.debezium.config.Field;
+import io.debezium.util.BoundedConcurrentHashMap;
+
+public class HeaderToValue<R extends ConnectRecord<R>> implements Transformation<R> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HeaderToValue.class);
+    public static final String FIELDS_CONF = "fields";
+    public static final String HEADERS_CONF = "headers";
+    public static final String OPERATION_CONF = "operation";
+    private static final String MOVE_OPERATION = "move";
+    private static final String COPY_OPERATION = "copy";
+    private static final int CACHE_SIZE = 64;
+    public static final String NESTING_SEPARATOR = ".";
+    public static final String ROOT_FIELD_NAME = "payload";
+
+    enum Operation {
+        MOVE(MOVE_OPERATION),
+        COPY(COPY_OPERATION);
+
+        private final String name;
+
+        Operation(String name) {
+            this.name = name;
+        }
+
+        static Operation fromName(String name) {
+            switch (name) {
+                case MOVE_OPERATION:
+                    return MOVE;
+                case COPY_OPERATION:
+                    return COPY;
+                default:
+                    throw new IllegalArgumentException();
+            }
+        }
+
+        public String toString() {
+            return name;
+        }
+    }
+
+    public static final Field HEADERS_FIELD = Field.create(HEADERS_CONF)
+            .withDisplayName("Header names list")
+            .withType(ConfigDef.Type.LIST)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .withDescription("Header names in the record whose values are to be copied or moved to record value.")
+            .required();
+
+    public static final Field FIELDS_FIELD = Field.create(FIELDS_CONF)
```

Comment:
@mfvitale Thanks, I wonder one detail. Do we want to have two separate config params? IMHO it'd be a bit error prone. Maybe a single config param `<header>:<field>,...` would be better?

Reply:
@jpechane In this case I prefer to have it separated because it's something usual for similar SMT https://docs.confluent.io/platform/current/connect/transforms/headerfrom.html#example
WDYT?